src/http_workers/worker_pool.rs

```diff
@@ -0,0 +1,35 @@
+use std::sync::Arc;
+
+use tokio::sync::{mpsc, oneshot, Mutex};
+
+use super::{
+    worker::{Worker, WorkerHandle},
+    Work,
+};
+
+pub struct WorkerPool<W>
+where
+    W: Work,
+{
+    receiver: mpsc::Receiver<oneshot::Sender<W>>,
+}
+
+impl<W> WorkerPool<W>
+where
+    W: Work + Send + Sync + Clone + 'static,
+{
+    pub async fn new(size: usize) -> WorkerPool<W> {
+        let (sender, receiver) = mpsc::channel(size);
```

Comment:
Size of `channel` buffer should be `1` as explained in the call. If it's still not clear why let me know and I will try to explain again.